org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/model/ModelLine.java

```diff
@@ -36,6 +36,10 @@ public List<TMToken> getTokens() {
 		return tokens;
 	}
 
+	public boolean isValid() {
```

Comment:
I actually don't like much the name `isValid` here. Indeed, in such an approach, being "invalid" is semantically precise as "state is not ready", it's actually not the opposite of valid. So if we can avoid adding this method, I think it keeps things clearer for consumers.

Reply:
I also thought about this. I think what invalid means is that the contained information is out of date / out of sync with the actual line in the editor. So we could change it to something like isOutdated / isSynced / isUpToDate, etc. any preference?

Reply:
We already have `isInvalid()` which is pretty explicit. Do we need a negated variant at all? If yes, then I thing `isUpToDate` is the closest meaning.

Reply:
I have no hard feelings for either getter. I can remove one. I just found `line.isValid()` nicer to use than having to use `!line.isInvalid()` which is a double negation.

Reply:
Our software world abuse the "invalid" term so it's not really a negation but a precise description of an actual state unlike "valid" ;) But I'm totally fine with @angelozerr suggestion of renaming `isInvalid` to `isDirty` if it helps.
My only concern is more that I don't like `isValid`.